org.eclipse.languageserver/src/org/eclipse/languageserver/ProjectSpecificLanguageServerWrapper.java

```diff
@@ -103,10 +86,39 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 			doc.setUri(fileURI.toString());
 			doc.setVersion(version);
 			this.change.setTextDocument(doc);
-			TextDocumentContentChangeEventImpl changeEvent = new TextDocumentContentChangeEventImpl();
-			changeEvent.setText(event.getDocument().get()); // TODO set to value after change
+			// create change event
+			TextDocumentContentChangeEventImpl changeEvent = toChangeEvent(event);
 			this.change.setContentChanges(Arrays.asList(new TextDocumentContentChangeEventImpl[] { changeEvent }));
 		}
+
+		/**
+		 * Convert Eclipse {@link DocumentEvent} to LS
+		 * {@link TextDocumentContentChangeEventImpl}.
+		 * 
+		 * @param event
+		 *            Eclipse {@link DocumentEvent}
+		 * @return the converted LS {@link TextDocumentContentChangeEventImpl}.
+		 */
+		private TextDocumentContentChangeEventImpl toChangeEvent(DocumentEvent event) {
```

Comment:
That's a nice change. Do you think this method would fit in the LSPEclipseUtils class, which is currently storing all those Eclipse <-> LSAPI transformations?